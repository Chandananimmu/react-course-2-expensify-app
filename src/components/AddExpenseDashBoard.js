import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { addExpense } from "../actions/expenses";
export class AddExpenseDashBoard extends React.Component {
  onSubmit = (expense) => {
    this.props.addExpense(expense);
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <p>This msg is from add dashboard </p>
        <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}
// const AddExpenseDashBoard = props => (
//   <div>
//     <p>This msg is from add dashboard </p>
//     <ExpenseForm
//       onSubmit={expense => {
//         //  console.log(expense);
//         props.onsubmit(expense);
//         props.history.push("/");
//       }}
//     />
//   </div>
// );
const mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => dispatch(addExpense(expense))
});
export default connect( undefined, mapDispatchToProps) (AddExpenseDashBoard);
 


