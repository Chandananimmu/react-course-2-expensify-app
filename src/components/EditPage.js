import React from "react";
import ExpenseForm from "./ExpenseForm";
import { connect } from "react-redux";
import { editExpense } from "../actions/expenses";
import { removeExpense } from "../actions/expenses";
export class EditPage extends React.Component {
  onSubmit =( expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };
  onClick = () => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
        <button onClick={this.onClick}>Remove</button>
      </div>
    );
  }
}
// const EditPage = props => (
//   <div>
//     <ExpenseForm
//     expense={props.expense}
//       onSubmit={(expense) => {
//        const id=props.expense.id
//         props.dispatch(editExpense(id,expense));
//         props.history.push("/")
//        // console.log("updated", expense);
//       }}
//     />
//      <button
//       onClick={() => {
//         const id=props.expense.id
//         props.dispatch(removeExpense({id}));
//         props.history.push("/")
//       }}
//     >
//       Remove
//     </button>
//   </div>
// );

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(
      expense => expense.id === props.match.params.id
    )
  };
};
const mapDispatchToProps = (dispatch, props) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: id => dispatch(removeExpense(id))
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditPage);
