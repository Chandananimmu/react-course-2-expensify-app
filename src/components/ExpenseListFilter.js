import React from "react";
import { connect } from "react-redux";
import { DateRangePicker } from "react-dates";
import {
  setFilterText,
  sortByDate,
  sortByAmount,
  setStartDate,
  setEndDate
} from "../actions/filters";

export class ExpenseListFilter extends React.Component {
  state = {
    calenderFocused: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = calenderFocused => {
 
    this.setState(() => ({
      calenderFocused
    }));
  };
  onTextChange = e => {
    this.props.setFilterText({ text: e.target.value });
  };
  onSortChange = e => {
    e.target.value === "date"
      ? this.props.sortByDate()
      : this.props.sortByAmount();
  };
  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          // onChange={e => {
          //   this.props.dispatch(setFilterText({ text: e.target.value }));
          // }}
          onChange={this.onTextChange}
        />
        <select
          value={this.props.filters.sortBy}
          // onChange={e => {
          //   this.props.dispatch(
          //     e.target.value === "date" ? sortByDate() : sortByAmount()
          //   );
          // }}
          onChange={this.onSortChange}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker 
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calenderFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
          
        />
      </div>
    );
  }
}
//const ExpenseListFilter = props =>
const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};
const mapDispatchToProps = dispatch => ({
  setStartDate: startDate => {
    dispatch(setStartDate(startDate));
  },
  setEndDate: endDate => {
    dispatch(setEndDate(endDate));
  },
  setFilterText: text => {
    dispatch(setFilterText(text));
  },
  sortByDate: () => {
    dispatch(sortByDate());
  },
  sortByAmount: () => {
    dispatch(sortByAmount());
  }
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ExpenseListFilter);
