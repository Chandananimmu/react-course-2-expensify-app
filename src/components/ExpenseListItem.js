import React from "react";
//import { connect } from "react-redux";
import { removeExpense } from "../actions/expenses";
import { Link } from "react-router-dom";
export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3> {description}</h3>
    </Link>

    <p> {amount}</p>
    <p>{createdAt}</p>
  </div>
);

// const mapDispatchFromProps = (push,dispatch) => ({
//  removeExpense: (id)=>dispatch(removeExpense({id})),

// })

//export default connect(null, mapDispatchFromProps)(ExpenseListItem);
export default ExpenseListItem;
