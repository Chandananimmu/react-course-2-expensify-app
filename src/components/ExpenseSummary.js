import React from "react";
import { connect } from "react-redux";
import getVisibleExpenses from "../selectors/expenses";
import getExpensesTotal from "../selectors/expenses-total";
import numeral from "numeral";
export const ExpenseSummary = props => (
  <div>
    <p>
      Viewing {props.expenses.length} expenses totalling{" "}
      {numeral(getExpensesTotal(props.expenses) / 100).format("$0,0.00")}
    </p>
  </div>
);
const mapToStateProps = state => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters),
  };
};
export default connect(mapToStateProps)(ExpenseSummary);
