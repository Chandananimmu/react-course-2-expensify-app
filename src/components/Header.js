import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to={`/create`} activeClassName="is-active">
      Go to create
    </NavLink>

    <NavLink to={`/help`} activeClassName="is-active">
      Go to help
    </NavLink>
    <NavLink to={`/`} exact={true} activeClassName="is-active">
      Go to home
    </NavLink>
  </header>
);
export default Header;
