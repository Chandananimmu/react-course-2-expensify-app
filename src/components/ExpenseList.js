import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import getVisibleExpenses from "../selectors/expenses";
export const ExpenseList = props => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Expenses</div>
      <div className="show-for-desktop">Expense</div>
      <div className="show-for-desktop">Amount</div>
  
</div>
<div className="list-body">
    {props.expenses.length === 0 ? (
      <div>
         <span className="list-item list-item--message">No expenses</span>
      </div>
     
    ) : (
      props.expenses.map(expense => {
        return <ExpenseListItem key={expense.id} {...expense} />;
      })
    )}
    </div>
  </div>
);
const mapToStateProps = state => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters)
  };
};

//const ConnnectedList = connect(mapToStateProps)(ExpenseList);
export default connect(mapToStateProps)(ExpenseList);
