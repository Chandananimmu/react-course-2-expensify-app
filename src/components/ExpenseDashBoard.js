import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilter from "./ExpenseListFilter";
import ExpenseSummary from "./ExpenseSummary";
const ExpenseDashBoard = () => (
  <div>
    <ExpenseSummary />
    <ExpenseListFilter />
    <ExpenseList />
  
  </div>
);
export default ExpenseDashBoard;
