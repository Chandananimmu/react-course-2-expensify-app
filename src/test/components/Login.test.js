import React from "react";
import { shallow } from "enzyme";
import {LoginPage} from "../../components/Login";
test("shud render login correctly", () => {
  const wrapper = shallow(<LoginPage startLogin={()=>{}} />);
  expect(wrapper).toMatchSnapshot();
});
test("shud logout on button click",()=>{
  const   startLogin = jest.fn();
  const  wrapper=shallow(<LoginPage startLogin={startLogin}/>);
  wrapper.find("button").simulate("click");
  expect(startLogin).toHaveBeenCalled();
  expect(wrapper).toMatchSnapshot();
})
