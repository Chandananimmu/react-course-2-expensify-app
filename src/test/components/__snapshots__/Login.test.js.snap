// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shud logout on button click 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;

exports[`shud render login correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
